include/zephyr/net/phy.h

```diff
@@ -74,6 +74,9 @@ typedef void (*phy_callback_t)(const struct device *dev,
  * public documentation.
  */
 __subsystem struct ethphy_driver_api {
+	/** Perform a soft reset */
+	int (*soft_reset)(const struct device *dev);
```

Comment:
Linux just calls this `reset`:
https://github.com/torvalds/linux/blob/master/include/linux/phy/phy.h#L123